### PR TITLE
Add JGithubCommits class

### PR DESCRIPTION
**Pull request by mbabker**
Added class to work with GitHub commits API per:
- http://developer.github.com/v3/repos/commits/
- http://developer.github.com/v3/git/commits/

### CODE REVIEW

File: libraries/joomla/github/commits.php [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+	 * @param   array   $parents  Array of the SHAs of the commits that were the parents of this commit.
+	 *                            If omitted or empty, the commit will be written as a root commit.
+	 *                            For a single parent, an array of one SHA should be provided.
+	 *                            For a merge commit, an array of more than one should be provided.
+	 *
+	 * @return  object
+	 *
+	 * @since   12.1
+	 */
+	public function create($user, $repo, $message, $tree, $parents = array())
```
Comment (realityking) [inline libraries/joomla/github/commits.php]: Since the rest of the package is type hinted so should probably $parents.

**Comment by mbabker** [comment]
The 3 additional checkstyle warnings are coming from #838 which looks intentional (they match the errors in the memcache driver).

**Comment by LouisLandry** [comment]
Very nice Michael... very nice.

**Comment by oc666** [comment]
Hey
I'm wondering what is used for?
Thanks

**Comment by mbabker** [comment]
The GitHub package is a group of classes that can be used in Joomla! code to interact with GitHub's API's.  The most notable use of the package right now is the pull tester (the @joomla-jenkins comments in pull requests).

This specific class doesn't have an implementation in code right now, but it adds the possibility to interact with the GitHub API's commits data points.

**Comment by oc666** [comment]
Michael, Thanks for reply

**Pull request merged**